[EDU-2289] fix: review marketplace guides (#421)
aziontech/docs: src/content/docs/en/pages/guides/marketplace/integrations/ab-testing-marketplace/index.mdx

---
title: How to install the A/B Testing integration through Azion Marketplace
description: How to use the A/B Testing integration from Marketplace.
meta_tags: marketplace, a/b test, how-to
namespace: documentation_how_to_ab_testing_configuration
permalink: /documentation/products/guides/ab-testing-marketplace/
---

Mainly, you can use **A/B tests** to validate certain addresses, interfaces, or even workflows. Tests enable you to experiment with variations and how they work with your users, providing two options to access certain content or complete task.

---

## Getting the integration 

To use the A/B test integration provided through Azion Marketplace, access [Real-Time Manager (RTM)](/en/documentation/products/guides/how-to-access-rtm/) and go to **Marketplace**.

> On the Marketplace's homepage, select the integration's card.

Reviewer: ```suggestion
On the Marketplace homepage, select the integration's card.
```

Once the integration's page opens, click the **Get It Now** button, at the bottom-right corner of the page, and a successful message appears to confirm your integration is installed.

:::tip

> You can search any integration by browsing through the cards or typing a keyword in the search box. 

Reviewer: ```suggestion
You can search any integration by browsing through the cards or typing a keyword in the search bar. 
```

8. Pass the URL you want to register for this origin.
9. Click the **Save** button.

After you save it, you'll be able to see the `Origin Key` generated by RTM for the origin you've registered. This will be used to fill the `originid` field on the **Arguments** box of the function.

:::note
You must register at least two origins.
:::

---

## Instantiating the integration

Still on the page of the application, follow these steps:

1. Select the **Functions** tab.
2. Click the **Add Function** button.
3. Choose an easy to remember name for your function.
4. On the dropdown function menu, select the **A/B Testing [Global]** function.

> This action will load the function, showing a form with the function’s code and, just above it, two tabs: **Code** and **Arguments**.

Reviewer 2: ```suggestion
This action will load the function, showing a form with the function's code and, just above it, two tabs: **Code** and **Arguments**.
```

On the **Code** tab, you’ll be able to navigate through the code, but won’t be able to change it. On the **Arguments** tab, you’ll pass the parameters.

Select the **Arguments** tab to open the form where you can add the parameters to execute your application.

:::note

> The parameters have divisions of at least three sections: `params`, `a`, `b`.

Reviewer: ```suggestion
The parameters have divisions of at least three sections: `params`, `a`, and `b`.
```

Where:

| Variable | Description |
|---|---|
| `name` | The name of the cookie you'll choose. Give a unique and easy-to-remember name |
| `expiration` | The expiration date of the cookie |
| `max_age` | The time, in seconds, for the lifespan of the cookie  |
| `domain` | The domain where you're running the tests |
| `path` | The subdomain to the cookie. You can use `/` to set the subdomain to all the domains |

> On the second section, called `a`, you can add the parameters to the first test, and then repear for the rest of the tests:

Reviewer 2: ```suggestion
On the second section, called `a`, you can add the parameters to the first test, and then repeat it for the rest of the tests:
```

| Variable | Description |
|---|---|
| `cookie_value` | A value to identify the cookie for this test variation |
| `prob` | The probability to send this address to the user. This value should be between `0` and `1`, with `0` being `0%` of chance and `1` being `100%` of chance |

> | `adressess_list` | The address for the a test. Example: `www.mytest_ab_of_my_site.com:443` |

Reviewer 2: ```suggestion
| `adressess_list` | The address for the `a` test. Example: `www.mytest_ab_of_my_site.com:443` |
```

| `originid` | The identification ID for the origin. This will be automatically generated by RTM |

> | `live_ingest` | It indicates if the origin is a live streaming. A Boolean value should be used here: `true` or `false` |

Reviewer 2: ```suggestion
| `live_ingest` | It indicates if the origin is a live streaming. A boolean value should be used here: `true` or `false` |
```

| `protocol_policy` | To force `HTTP` or `HTTPS`, or even keep the default value by passing the preserve value |
| `path` | It indicates the subdomain of the cookie. You can leave it blank, `""`, to indicate that it's valid for all domains |
| `cache_key` | A name to identify the cache. It should be unique and you can't repeat it elsewhere |
| `host` | The name of the host that the requisition sends |

---

## Configuring the Rules Engine

> In the **Rules Engine**, you'll pass the *criteria* and *behavior* to run the integration.

Reviewer 2: ```suggestion
In **Rules Engine**, you'll pass the *criteria* and *behavior* to run the integration.
```
